### PR TITLE
DB optimization 1

### LINKED ISSUE #3136: Database use improvements
Let's do some small database use improvements for a start.

- [ ] don't audit log static resources
- [ ] don't fetch user profile for language, prefer language cookie

### PR DESCRIPTION
**Pull request by Macroz**
Close #3136

- don't save to audit log static resources as that is not interesting
- prefer language cookie to fetching settings from db
- provide `context/*request*` for later use
- also clean up and prefer eager evaluation for better locality of database use

See also #3137

# Checklist for author

Remove items that aren't applicable, check items that are done.

## Reviewability
- [x] Link to issue

## Documentation
- [x] Update changelog if necessary

## Testing
- [x] Valuable features are integration / browser / acceptance tested automatically

## Follow-up
- [x] New tasks are created for pending or remaining tasks

### CODE REVIEW

File: test/clj/rems/db/test_core.clj
Comment (aatkin) [inline test/clj/rems/db/test_core.clj:1]: git apparently didn't recognize that the file was renamed/moved? so this file should be removed as it's empty now

File: test/clj/rems/db/test_catalogue.clj [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  (let [today (time/now)
+        yesterday (time/minus today (time/days 1))
+        all-items [{:id :normal
+                    :start nil
+                    :end nil}
+                   {:id :expired
+                    :start nil
+                    :end yesterday}]
+
+                                        ; the following idiom can be used when reading database entries with 'end' field
```
Comment (aatkin) [inline test/clj/rems/db/test_catalogue.clj:60]: comment indentation is off (would be nice if linter could catch these)

File: src/clj/rems/util.clj [hunk 2]
```diff
@@ -110,3 +110,11 @@
   [dir]
   (.mkdirs dir)
   (delete-directory-contents-recursively dir))
+
+(defn contains-all-kv-pairs? [supermap map]
+  (superset? (set supermap) (set map)))
+
+(defn apply-filters [filters coll]
+  (let [filters (or filters {})]
+    (filter #(contains-all-kv-pairs? % filters) coll)))
```
Comment (aatkin) [inline src/clj/rems/util.clj:119]: these probably could exist in common namespace as they are usable from cljs too.

**Review by aatkin — APPROVED** [review]

File: test/clj/rems/db/test_catalogue.clj (hunk 1 above)
Comment (Macroz) [reply test/clj/rems/db/test_catalogue.clj:60]: Single semicolon is meant for end of line comments and the default behavior of Emacs is to indent it to the middle of the line. I guess the correct fix is to make this `;;` so it follows the code.

**Pull request merged**